### PR TITLE
Company signup - Split Individual/Company

### PR DESCRIPTION
**Pull request by alanhoskins**
[TICKET](https://artistguide.atlassian.net/browse/TICKET)

**Description:**

This is just the base set that splits the signup into and individual signup and company. Still working on this as I go. Company PR incoming soon.

**PR Checklist:**

- [X ] If you haven't made changes to `package.json`, `package-lock.json` is not present in this PR
- [X ] You have tested these changes locally by running the app and letting it build successfully
- [ X] Eslint ran successfully upon commit and `--force` was not used to force this PR

_Make sure you **squash and merge** rather than simply merging all commits at once_.

At least **1** PR approval is required by a Tech Lead to be merged. **2** approvals total are required.

### CODE REVIEW

File: src/components/SignUp/Group/index.tsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            {
+              basics_18_plus: basics18Plus,
+              first_name: basicsFirstName,
+              last_name: basicsLastName,
+              privacy_agreement: true,
+              uid: userId
+            }
+          );
+
+          const profileRef = await addDoc(
```
Comment (bestfoodalex) [inline src/components/SignUp/Group/index.tsx:82]: I wonder if we should have different firebase tables for theater profiles vs artist profiles

File: src/components/SignUp/Group/index.tsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+      basicsPassword
+    )
+      .then(async res => {
+        try {
+          const userId = res.user.uid;
+
+          const accountRef = await addDoc(
+            collection(firebaseFirestore, 'accounts'),
+            {
+              basics_18_plus: basics18Plus,
```
Comment (bestfoodalex) [inline src/components/SignUp/Group/index.tsx:74]: Does the theater sign-up profile creation include these same fields, like 18+?

File: src/components/SignUp/Group/index.tsx [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    createDefaultStepErrorsObj(flatSteps(defaultSteps))
+  );
+
+  // console.log('formData', { formData });
+  // defaults for our defaultSteps
+  const { step, index, navigation } = useStep({
+    initialStep: currentStep,
+    steps: defaultSteps
+  });
+  const stepId = (step as Step).id;
```
Comment (bestfoodalex) [inline src/components/SignUp/Group/index.tsx:38]: I feel like we can extract some of this logic we have in all of our SignUp steps like individual 1, individual 2, and now group and some generic utilities for it 🤔

File: src/components/SignUp/Group/index.tsx [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+  // based on which step we're on, return a different step component and pass it the props it needs
+  const stepFrame = () => {
+    // console.log('stepFrame', stepId);
+    const props = { formData, setForm, navigation };
+    let returnStep;
+
+    switch (stepId) {
+      case 'basics':
+        returnStep = (
+          <IndividualBasics
```
Comment (bestfoodalex) [inline src/components/SignUp/Group/index.tsx:123]: I'm confused - this looks like all of the same stuff as the Individual sign-up? Is this just temporary to get the new folder structure in first and then you'll go back and add the group sign-up steps?

File: src/pages/App.tsx [hunk 5]
```diff
@@ -70,7 +70,7 @@ const CAG = () => (
 );
 
 const App = () => {
-  const [currentUser, setCurrentUser] = useState<any>(null);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
```
Comment (bestfoodalex) [inline src/pages/App.tsx:73]: Thanks for adding some typing in for us 🙏🏻

File: src/pages/SignUp.tsx [hunk 6]
```diff
@@ -1,318 +1,67 @@
...
+import IndividualSignUp from '../components/SignUp/Individual';
+import {
+  ButtonCol,
+  PageFooterRow
+} from '../components/SignUp/SignUpFooterStyles';
+import Button from '../genericComponents/Button';
+
+const SignUp: React.FC = () => {
+  const [currentStep, setCurrentStep] = useState(-1);
+  const [accountType, setAccountType] = useState<'individual' | 'group' | null>(
```
Comment (bestfoodalex) [inline src/pages/SignUp.tsx:16]: We might want to create some types files for defining these in one place

File: src/pages/SignUp.tsx [hunk 7]
```diff
@@ -1,318 +1,67 @@
...
-      case 'profilePreview' as any:
-        returnStep = <Profile previewMode={true} />;
-        break;
-      default:
-        returnStep = <></>;
-        break;
-    }
+  }
+
+  if (accountType === 'individual') {
```
Comment (bestfoodalex) [inline src/pages/SignUp.tsx:49]: [nit] this may look better as a switch/case as well

**Review by bestfoodalex — APPROVED** [review]
Looking good as a base re-structuring of the files before you dive in with the group steps themselves!

File: src/components/SignUp/Group/index.tsx (hunk 4 above)
Comment (bestfoodalex) [reply src/components/SignUp/Group/index.tsx:123]: Oh, duh, that's what this PR is haha

File: src/components/SignUp/Group/index.tsx (hunk 1 above)
Comment (alanhoskins) [reply src/components/SignUp/Group/index.tsx:82]: I agree it should probably be a different table.

File: src/components/SignUp/Group/index.tsx (hunk 2 above)
Comment (alanhoskins) [reply src/components/SignUp/Group/index.tsx:74]: It won't. This was just an initial stub to get it to where it was before the split.

File: src/components/SignUp/Group/index.tsx (hunk 3 above)
Comment (alanhoskins) [reply src/components/SignUp/Group/index.tsx:38]: Yep I will extract more.

File: src/pages/SignUp.tsx (hunk 6 above)
Comment (alanhoskins) [reply src/pages/SignUp.tsx:16]: Agreed.

File: src/pages/SignUp.tsx (hunk 7 above)
Comment (alanhoskins) [reply src/pages/SignUp.tsx:49]: Yeah I will take a look at that too.

**Comment by alanhoskins** [comment]
Also quick question, would it be preferred to call this "company" or "group". I noticed both were used in some places and I wanted to find one and stick with it if possible.

**Comment by bestfoodalex** [comment]
@alanhoskins 

> Also quick question, would it be preferred to call this "company" or "group". I noticed both were used in some places and I wanted to find one and stick with it if possible.

Let me check with Steven and get back to you. I think we'll determine from a product perspective which is preferred, then match our naming in the code to that answer 👍🏻

**Pull request merged**